The backfill code for Dominant color

pbearne opened the pull request:
## Summary
this PR has code that will backfill the existing images with the Dominant colors
by default, it creates a run once call per image to add the image metadata if missing
this can be turned off via the filter "dominant_color_enable_back_fill" to be run as part of the page load with this filter "dominant_color_enable_back_fill_realtime"

<!-- Please reference the issue this PR fixes. If this PR does not fix the entire issue, change this to Addresses #... instead. -->
Fixes #282

## Relevant technical choices

<!-- Please describe your changes. -->

## Checklist

- [ ] PR has either `[Focus]` or `Infrastructure` label.
- [ ] PR has a `[Type]` label.
- [ ] PR has a milestone or the `no milestone` label.

pbearne committed:
alpha

we have code
I have no idea if the imagick code path works
not sure the js to remove bg works

pbearne committed:
alpha

we have code
I have no idea if the imagick code path works
not sure the js to remove bg works

pbearne committed:
added some tests

pbearne committed:
added some tests

Commit:
Added some more tests

Commit:
Added some more tests

Commit:
code review changes

Commit:
code review changes

pbearne committed:
removed iframe

pbearne committed:
removed iframe

Commit:
code review changes
created classes for image functions
Updated tests

NOT all tests are passsing

Commit:
code review changes
created classes for image functions
Updated tests

NOT all tests are passsing

Commit:
change filter to match core change

Commit:
change filter to match core change

pbearne committed:
Merge branch 'WordPress:trunk' into INITS-199_domiant_color

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Jonny Harris <spacedmonkey@users.noreply.github.com>

pbearne committed:
Merge branch 'WordPress:trunk' into INITS-199_domiant_color

pbearne committed:
Added test files

got get_has_transparency() working

pbearne committed:
Added test files

got get_has_transparency() working

pbearne committed:
comments

pbearne committed:
comments

pbearne committed:
fixed test

pbearne committed:
fixed test

pbearne committed:
fix linting errors

pbearne committed:
fix linting errors

pbearne committed:
more linting

pbearne committed:
more linting

pbearne committed:
the last linting

pbearne committed:
the last linting

pbearne committed:
removed old conditional code

pbearne committed:
removed old conditional code

pbearne committed:
Merge branch 'INITS-199_domiant_color' of github.com:xwp/performance into INITS-199_domiant_color

pbearne committed:
refactored to get the linting to work

pbearne committed:
refactored to get the linting to work

pbearne committed:
added filer to allow fine grain control
enable_dominant_color_for_image

pbearne committed:
added filer to allow fine grain control
enable_dominant_color_for_image

pbearne committed:
more linting fixes

pbearne committed:
more linting fixes

manuelRod committed:
lint fix and directory wording typo

manuelRod committed:
lint fix and directory wording typo

manuelRod committed:
adding some extra testing

manuelRod committed:
adding some extra testing

spacedmonkey committed:
Fix lints

spacedmonkey committed:
Fix lints

pbearne committed:
Delete phpinfo.php

pbearne committed:
Delete phpinfo.php

pbearne committed:
code review feedback fixes

pbearne committed:
code review feedback fixes

pbearne committed:
code review feedback fixes

pbearne committed:
code review feedback fixes

pbearne committed:
reverted lock

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

# Conflicts:
#	composer.lock

spacedmonkey committed:
Use inline styles

spacedmonkey committed:
Merge branch 'INITS-199_domiant_color' of github.com:xwp/performance into INITS-199_domiant_color

spacedmonkey committed:
Revert lock file

spacedmonkey committed:
Apply suggestions from code review

spacedmonkey committed:
Small tweaks

spacedmonkey committed:
Merge branch 'INITS-199_domiant_color' of github.com:xwp/performance into INITS-199_domiant_color

spacedmonkey committed:
Fix lint

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

# Conflicts:
#	composer.lock

pbearne committed:
added code to get and save dominant color if not already set in image meta

pbearne committed:
whitespaces

pbearne committed:
whitespaces

pbearne committed:
Merge branch 'WordPress:trunk' into INITS-199_domiant_color

pbearne committed:
Removed class from code

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

pbearne committed:
Removed class file

pbearne committed:
fixed tests

pbearne committed:
fixed $this ref

pbearne committed:
Stopped adding the filters not needed for WP version 6 onwards as we have a new filter (wp_content_img_tag) to use

pbearne committed:
White space tidy

pbearne committed:
Changed how we back-fill the previously uploaded images.
We now by default create run once cron tasks, so we don't slow the page load down

pbearne committed:
whitespace

pbearne committed:
Update modules/images/dominant-color/class-wp-image-editor-gd-with-color.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/class-wp-image-editor-gd-with-color.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/class-wp-image-editor-gd-with-color.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/class-wp-image-editor-imagick-with-color.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
removed the wp_ prefix
and few other changes from review

pbearne committed:
removed the backfill from this PR and

pbearne committed:
whitespace and test file rename

pbearne committed:
whitespace and test file rename

pbearne committed:
whitespace and test file rename

pbearne committed:
Fixed the tests

pbearne committed:
Fixed the version compare
changed code to always set has_transparency in meta so early bail works

pbearne committed:
more fixes for has_transparency changes

pbearne committed:
Use Yoda Condition checks, you must.

pbearne committed:
Added test for dominant_color_color_is_light and add code to handle short hex colors

pbearne committed:
added return early to the images class function if no image object set

pbearne committed:
added return early to the images class function if no image object set

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Eugene Manuilov <eugene.manuilov@gmail.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Eugene Manuilov <eugene.manuilov@gmail.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Eugene Manuilov <eugene.manuilov@gmail.com>

pbearne committed:
removed the second call to fetch editor in dominant_color_get_has_transparency()

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

# Conflicts:
#	modules/images/dominant-color/load.php

pbearne committed:
linting white spaces

pbearne committed:
added checks for methods before calling just in case ours class are overloaded

pbearne committed:
added checks for methods before calling just in case ours class are overloaded

pbearne committed:
added checks for methods before calling just in case ours class are overloaded

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

# Conflicts:
#	modules/images/dominant-color/load.php

pbearne committed:
changed method_exists to look at the class not its name for check

pbearne committed:
changed method_exists to look at the class not its name for check

pbearne committed:
white

pbearne committed:
whitespace

pbearne committed:
whitespace

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Eugene Manuilov <eugene.manuilov@gmail.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Eugene Manuilov <eugene.manuilov@gmail.com>

spacedmonkey commented:
@pbearne There are unit test and coding lint failures here. Can you please take a look into this.

pbearne committed:
added filter to override Default color

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

pbearne committed:
Update modules/images/dominant-color/load.php

change the code to append styles

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
Update modules/images/dominant-color/load.php

refactor class adding

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
Update modules/images/dominant-color/load.php

remove extra line

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
Update modules/images/dominant-color/load.php

remove extra line

Co-authored-by: Crisoforo Gaspar Hernández <hello@crisoforo.com>

pbearne committed:
whitespace fixes

pbearne committed:
whitespace fixes

pbearne committed:
Fixed test after changing default color

pbearne committed:
merged head in to backfill

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Update modules/images/dominant-color/load.php

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

pbearne committed:
Changed code to not store color if we get an error back from create functions
Moved the default color to the output and not store in meta

moved @since in docs blocks

pbearne committed:
Merge remote-tracking branch 'origin/INITS-199_domiant_color' into INITS-199_domiant_color

pbearne committed:
Merge branch 'INITS-199_domiant_color' into inits-199a-with_backfill

pbearne committed:
whitespace

pbearne committed:
whitespace

pbearne committed:
added a couple of missed @since

pbearne committed:
combined dominant_color_has_transparency_metadata into dominant_color_metadata to save duplicate code

pbearne commented:
not needed
